packages/@aws-cdk/aws-ecs/lib/load-balanced-service.ts

```diff
@@ -0,0 +1,133 @@
+import { ICertificate } from '@aws-cdk/aws-certificatemanager';
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+import cdk = require('@aws-cdk/cdk');
+import { BaseService } from './base/base-service';
+import { ICluster } from './cluster';
+import { IContainerImage } from './container-image';
+
+export enum LoadBalancerType {
+  Application,
+  Network
+}
+
+export interface LoadBalancedServiceProps {
+  /**
+   * The cluster where your service will be deployed
+   */
+  cluster: ICluster;
+
+  /**
+   * The image to start.
+   */
+  image: IContainerImage;
+
+  /**
+   * The container port of the application load balancer attached to your Fargate service. Corresponds to container port mapping.
+   *
+   * @default 80
+   */
+  containerPort?: number;
+
+  /**
+   * Determines whether the Application Load Balancer will be internet-facing
+   *
+   * @default true
+   */
+  publicLoadBalancer?: boolean;
+
+  /**
+   * Number of desired copies of running tasks
+   *
+   * @default 1
+   */
+  desiredCount?: number;
+
+  /**
+   * Whether to create an application load balancer or a network load balancer
+   * @default application
+   */
+  loadBalancerType?: LoadBalancerType
+
+  /**
+   * Environment variables to pass to the container created for the service
+   * @default No environment variables
+   */
+  containerEnvironment?: {[key: string]: string};
+
+  /**
+   * Certificate Manager certificate to associate with the load balancer.
+   * Setting this option will set the load balancer port to 443.
+   */
+  certificate?: ICertificate;
+}
+
+/**
+ * Base class for load-balanced Fargate and ECS service
+ */
+export abstract class LoadBalancedService extends cdk.Construct {
```

Comment:
Please rename to `LoadBalancedServiceBase` to make it clear from the name that it's not meant to be instantiated. I know, it's `abstract` already, but more clarity never hurts.